### DISCUSSION TITLE
CIP 21 - Governable Lookback Window - Discussions

### DISCUSSION THREAD

**Discussion by YazzyYaz** [celo-org/celo-proposals#330] [CIP Discussions]
- [CIP](https://github.com/celo-org/celo-proposals/blob/master/CIPs/cip-0021.md)
- [Monorepo](https://github.com/celo-org/celo-monorepo/pull/4747)
- [Blockchain](https://github.com/celo-org/celo-blockchain/pull/1136)

**Deployment Checklist** 
- [ ] Specification
- [X] Implementation
    - [X] celo-blockchain
    - [X] contracts in celo-monorepo
- [ ] Testing
    - [ ]  TODO: @mcortesi + @andres-dg 
- [ ] Pre-deploy
    - [ ] Governance contract upgrade deployed
        - [ ] Alfajores
        - [ ] Baklava
        - [ ] Mainnet
- [ ] Verification
    - [ ] Take a validator offline, ensure that the lookback is calculated correctly

**Comment by prestwich** [comment]
@andres-dg 

re [backwards compatibility](https://github.com/celo-org/celo-proposals/blob/master/CIPs/cip-0021.md#backwards-compatibility) what is the behavior if the contract has not yet been deployed? Based on the implementation, it appears that this would be a fatal error?
